jdaviz/configs/specviz2d/plugins/cross_dispersion_profile/cross_dispersion_profile.py

```diff
@@ -0,0 +1,358 @@
+import math
+
+import astropy.units as u
+import numpy as np
+from bqplot import LinearScale
+from specreduce.tracing import FlatTrace
+from specreduce.utils import measure_cross_dispersion_profile
+from traitlets import Bool, Float, Integer, List, Unicode, observe
+
+from jdaviz.core.events import GlobalDisplayUnitChanged
+from jdaviz.core.marks import PluginLine, PluginScatter
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import (DatasetSelect, PluginTemplateMixin,
+                                        PlotMixin)
+from jdaviz.core.unit_conversion_utils import (all_flux_unit_conversion_equivs,
+                                               flux_conversion_general)
+from jdaviz.core.user_api import PluginUserApi
+
+__all__ = ['CrossDispersionProfile']
+
+
+@tray_registry('cross-dispersion-profile', label="Cross Dispersion Profile")
+class CrossDispersionProfile(PluginTemplateMixin, PlotMixin):
+    """
+    The Cross Dispersion Profile plugin allows for visualizaion of the
+    cross-dispersion profile of 2d spectra, at a specified wavelength / pixel
+    and window.
+
+    The following attributes and methods are available through the
+    :ref:`public plugin API <plugin-apis>`:
+
+    * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.show`
+    * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.open_in_tray`
+    * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.close_in_tray`
+    * :meth:`measure_cross_dispersion_profile`
+    * ``dataset``
+        Dataset used to calculate and plot cross-dispersion profile.
+    * ``pixel``
+        Pixel on spectral axis used to compute and plot profile.
+    * ``y_pixel``
+        Center of profile on cross-dispersion axis.
+    * ``use_full_width``
+        If true, full cross-dispersion axis will be used to compute the profile.
+    * ``width``
+        If use_full_with is False, this value will set the size of the window
+        for the profile on the cross-dispersion axis, centered at y_pixel.
+    * ``profile``
+        Computed profile, as a Quantity array.

```

Comment:
does it make sense to expose both `measure_cross_dispersion_profile` and `profile` (since the method returns the profile)?  Either way, please let's make sure the list here matches what is exposed in `user_api` before merging

Reply:
(now we just need `profile` removed from the API docs here)